src/features/jserrors/aggregate/index.js

```diff
@@ -197,98 +197,94 @@ export class Aggregate extends AggregateBase {
+    // They each will also tack on their respective properties to the params object as part of the decision flow.
+    if (softNavInUse) handle('jserror', [params, time], undefined, FEATURE_NAMES.softNav, this.ee)
+    else handle('errorAgg', jsErrorEvent, undefined, FEATURE_NAMES.spa, this.ee)
+
+    if (params.browserInteractionId && !params._softNavFinished) { // hold onto the error until the in-progress interaction is done, eithered saved or discarded
+      this.bufferedErrorsUnderSpa[params.browserInteractionId] ??= []
+      this.bufferedErrorsUnderSpa[params.browserInteractionId].push(jsErrorEvent)
+    } else if (params._interactionId != null) { // same as above, except tailored for the way old spa does it
+      this.bufferedErrorsUnderSpa[params._interactionId] = this.bufferedErrorsUnderSpa[params._interactionId] || []
+      this.bufferedErrorsUnderSpa[params._interactionId].push(jsErrorEvent)
+    } else {
+      // Either there is no interaction (then all these params properties will be undefined) OR there's a related soft navigation that's already completed.
+      // The old spa does not look up completed interactions at all, so there's no need to consider it.
+      this.#storeJserrorForHarvest(jsErrorEvent, params.browserInteractionId !== undefined, params._softNavAttributes)
     }
   }
 
-  onInteractionSaved (interaction) {
-    if (!this.errorCache[interaction.id] || this.blocked) return
+  #storeJserrorForHarvest (errorInfoArr, softNavOccurredFinished, softNavCustomAttrs = {}) {
+    let [type, bucketHash, params, newMetrics, localAttrs] = errorInfoArr
+    const allCustomAttrs = {}
 
-    this.errorCache[interaction.id].forEach((item) => {
-      var customParams = {}
-      var globalCustomParams = item[4]
-      var localCustomParams = item[5]
+    if (softNavOccurredFinished) {
+      Object.entries(softNavCustomAttrs).forEach(([k, v]) => setCustom(k, v)) // when an ixn finishes, it'll include stuff in jsAttributes + attrs specific to the ixn
+      bucketHash += params.browserInteractionId
 
-      mapOwn(globalCustomParams, setCustom)
-      mapOwn(interaction.root.attrs.custom, setCustom)
-      mapOwn(localCustomParams, setCustom)
-
-      var params = item[2]
-      params.browserInteractionId = interaction.root.attrs.id
-      delete params._interactionId
-
-      if (params._interactionNodeId) {
-        params.parentNodeId = params._interactionNodeId.toString()
-        delete params._interactionNodeId
-      }
-
-      var hash = item[1] + interaction.root.attrs.id
-      var jsAttributesHash = stringHashCode(stringify(customParams))
-      var aggregateHash = hash + ':' + jsAttributesHash
+      delete params._softNavAttributes // cleanup temp properties from synchronous evaluation; this is harmless when async from soft nav (properties DNE)
+      delete params._softNavFinished
```

Comment:
is this stuff duplicative to the `#storeJserrorForHarvest` arguments of `softNavCustomAttrs = {}`?

Reply:
No, the params are sent as is AFAICT(ell). These need to be scrubbed so they're not sent under the errors.